test/support/require_cookie_plug.ex

```diff
@@ -0,0 +1,62 @@
+defmodule PhoenixTest.Plugs.RequireCookiePlug do
+  @moduledoc """
+  Allows conn to pass if it has the correct cookie.
+
+  If no cookie, or invalid cookie, returns 403.
+  """
+  import Phoenix.Controller, only: [protect_from_forgery: 1]
+  import Plug.Conn
+
+  @encrypted_cookie "encrypted_cookie"
+  @signed_cookie "signed_cookie"
+  @plain_cookie "plain_cookie"
+
+  def cookie_name(:encrypted), do: @encrypted_cookie
+  def cookie_name(:signed), do: @signed_cookie
+  def cookie_name(:plain), do: @plain_cookie
+
+  def cookie_options(:encrypted) do
+    [sign: false, encrypt: true, same_site: "Strict", http_only: true, secure: true]
+  end
+
+  def cookie_options(:signed) do
+    [sign: true, encrypt: false, same_site: "Lax", http_only: false, secure: false]
+  end
+
+  def cookie_options(:plain) do
+    [sign: false, encrypt: false, same_site: "Lax", http_only: false, secure: false]
+  end
+
+  @doc """
+  Return a cookie value that will pass through the plug's check.
+
+  A cookie value must be a binary unless the cookie is signed/encrypted.
+  """
+  def valid_cookie_value(:plain), do: "the secret is mighty_boosh"
+  def valid_cookie_value(_), do: %{secret: "mighty_boosh"}
+
+  def init(_opts), do: nil
+
+  def call(conn, _opts) do
+    conn
+    |> fetch_cookies(signed: [@signed_cookie], encrypted: [@encrypted_cookie])
+    |> protect_from_forgery()
+    |> ensure_correct()
+  end
+
+  defp ensure_correct(conn) do
+    case Map.get(conn.cookies, @encrypted_cookie) || Map.get(conn.cookies, @signed_cookie) ||
+           Map.get(conn.cookies, @plain_cookie) do
+      %{secret: "mighty_boosh"} ->
+        conn
+
+      "the secret is mighty_boosh" ->
+        conn
+
+      _ ->
+        conn
+        |> put_status(403)
+        |> halt()
+    end
+  end
```

Comment:
I would prefer to see this assertion in the actual test itself.

While the test suite is comprehensive, there is a little too much indirection for my taste.
I prefer more explicit tests at the cost of more redundant code in the tests.

Reply:
I don't understand. Can you elaborate on what you mean by "assertion in the actual test itself"?

Reply:
Sorry, I explained that badly.
I pushed a suggestion in `a2ff53`, that probably explains it better :()